stonesoup/models/measurement/tests/test_models.py

```diff
@@ -15,117 +16,53 @@
 
     # Get rotation matrix
     theta_z = - rotation_offset[2, 0]
-    cos_z, sin_z = np.cos(theta_z), np.sin(theta_z)
-    rot_z = np.array([[cos_z, -sin_z, 0],
-                      [sin_z, cos_z, 0],
-                      [0, 0, 1]])
-
     theta_y = - rotation_offset[1, 0]
-    cos_y, sin_y = np.cos(theta_y), np.sin(theta_y)
-    rot_y = np.array([[cos_y, 0, sin_y],
-                      [0, 1, 0],
-                      [-sin_y, 0, cos_y]])
-
     theta_x = - rotation_offset[0, 0]
-    cos_x, sin_x = np.cos(theta_x), np.sin(theta_x)
-    rot_x = np.array([[1, 0, 0],
-                      [0, cos_x, -sin_x],
-                      [0, sin_x, cos_x]])
-
-    rotation_matrix = rot_z@rot_y@rot_x
 
+    rotation_matrix = rotz(theta_z) @ roty(theta_y) @ rotx(theta_x)
     xyz_rot = rotation_matrix @ xyz
-    x = xyz_rot[0, 0]
-    y = xyz_rot[1, 0]
-    # z = 0  # xyz_rot[2, 0]
 
-    rho = np.sqrt(x**2 + y**2)
-    phi = np.arctan2(y, x)
+    rho, phi, _ = cart2sphere(*xyz_rot[:, 0])
 
     return np.array([[Bearing(phi)], [rho]])
 
 
-def h3d(state_vector,  translation_offset, rotation_offset):
-
-    xyz = [[state_vector[0, 0] - translation_offset[0, 0]],
-           [state_vector[1, 0] - translation_offset[1, 0]],
-           [state_vector[2, 0] - translation_offset[2, 0]]]
+def h3d(state_vector, pos_map,  translation_offset, rotation_offset):
+    xyz = state_vector[pos_map, :] - translation_offset
 
     # Get rotation matrix
     theta_z = - rotation_offset[2, 0]
-    cos_z, sin_z = np.cos(theta_z), np.sin(theta_z)
-    rot_z = np.array([[cos_z, -sin_z, 0],
-                      [sin_z, cos_z, 0],
-                      [0, 0, 1]])
-
     theta_y = - rotation_offset[1, 0]
-    cos_y, sin_y = np.cos(theta_y), np.sin(theta_y)
-    rot_y = np.array([[cos_y, 0, sin_y],
-                      [0, 1, 0],
-                      [-sin_y, 0, cos_y]])
-
     theta_x = - rotation_offset[0, 0]
-    cos_x, sin_x = np.cos(theta_x), np.sin(theta_x)
-    rot_x = np.array([[1, 0, 0],
-                      [0, cos_x, -sin_x],
-                      [0, sin_x, cos_x]])
-
-    rotation_matrix = rot_z@rot_y@rot_x
 
+    rotation_matrix = rotz(theta_z) @ roty(theta_y) @ rotx(theta_x)
     xyz_rot = rotation_matrix @ xyz
-    x = xyz_rot[0, 0]
-    y = xyz_rot[1, 0]
-    z = xyz_rot[2, 0]
 
-    rho = np.sqrt(x**2 + y**2 + z**2)
-    phi = np.arctan2(y, x)
-    theta = np.arcsin(z/rho)
+    rho, phi, theta = cart2sphere(*xyz_rot[:, 0])
 
     return np.array([[Elevation(theta)], [Bearing(phi)], [rho]])
 
 
-def hbearing(state_vector, translation_offset, rotation_offset):
-    xyz = [[state_vector[0, 0] - translation_offset[0, 0]],
-           [state_vector[1, 0] - translation_offset[1, 0]],
-           [state_vector[2, 0] - translation_offset[2, 0]]]
+def hbearing(state_vector, pos_map, translation_offset, rotation_offset):
+    xyz = state_vector[pos_map, :] - translation_offset
 
     # Get rotation matrix
     theta_z = - rotation_offset[2, 0]
```

Comment:
Marginally cleaner as 
```suggestion
    theta_x, theta_y, theta_z = - rotation_offset[:, 0]
```

Reply:
done